File: gestion-tickets/src/app/_models/TicketFilterRequest.ts
export interface TicketFilterRequest {
  userId?: number;
  ownerId?: number;
  personnelId?: number;
  projetId?: number; 
  start?: string;    
  end?: string;
  granularity: 'daily' | 'weekly' | 'monthly' | 'yearly' | 'none';
}

File: GestionTicketsAPI/Helpers/TicketFilterRequest.cs
public class TicketFilterRequest
{
    public int? UserId      { get; set; }
    public int? OwnerId { get; set; }
    public int? PersonnelId { get; set; }
    public int? ProjetId { get; set; }
    public DateTime? Start { get; set; }
    public DateTime? End    { get; set; }
    public string  Granularity { get; set; } = "monthly";
}
